supervision/metrics/detection.py

```diff
@@ -456,3 +460,329 @@ def plot(
                 save_path, dpi=250, facecolor=fig.get_facecolor(), transparent=True
             )
         return fig
+
+
+@dataclass(frozen=True)
+class MeanAveragePrecision:
+    """
+    Mean Average Precision for object detection tasks.
+
+    Attributes:
+        map (float): mAP value.
+        map50 (float): mAP value at IoU `threshold = 0.5`.
+        map75 (float): mAP value at IoU `threshold = 0.75`.
+        average_precisions (np.ndarray): values for every classes.
```

Comment:
@hardikdava could you help me out here? What is stored under the `average_precisions` field?

Reply:
Average precision holds the result of ap at different iou threshold starting from 0.5 to 0.95 at an interval of 0.5 also the results are for each class. Columns represents classes and rows represents iou threshold.

Reply:
So what would be the expected shape of this np array? (N, 10)? Where N is the amount of classes?

Reply:
Yeah, numpy array of (num of class, 10).

Reply:
I Done some unit teats and it looks like the shape is different :/

Reply:
### Test 1

```python
>>> import supervision as sv
>>> import numpy as np

>>> predictions = [
...     np.empty((0, 6), dtype=np.float32),
... ]
>>> targets = [
...     np.empty((0, 5), dtype=np.float32),
... ]

>>> sv.MeanAveragePrecision.from_tensors(predictions=predictions, targets=targets)
# MeanAveragePrecision(map=0, map50=0, map75=0, average_precisions=[])
```

### Test 2

```python
>>> import supervision as sv
>>> import numpy as np

>>> predictions = [
...     np.array([
...         [0.0, 0.0, 1.0, 1.0, 1, 0.9],
...     ]),
... ]
>>> targets = [
...     np.array([
...         [0.0, 0.0, 1.0, 1.0, 1],
...     ]),
... ]

>>> sv.MeanAveragePrecision.from_tensors(predictions=predictions, targets=targets)
# MeanAveragePrecision(map=0.9949999999999999, map50=0.995, map75=0.995, average_precisions=array([      0.995]))
```

### Test 3

```python
>>> import supervision as sv
>>> import numpy as np

>>> predictions = [
...     np.array([
...         [0.0, 0.0, 1.0, 1.0, 1, 0.9],
...     ]),
...     np.array([
...         [0.0, 0.0, 1.0, 1.0, 1, 0.9],
...     ]),
...     np.array([
...         [0.0, 0.0, 1.0, 1.0, 2, 0.9],
...     ]),
... ]
>>> targets = [
...     np.array([
...         [0.0, 0.0, 1.0, 1.0, 1],
...     ]),
...     np.array([
...         [0.0, 0.0, 0.8, 0.8, 1, 0.9],
...     ]),
...     np.array([
...         [0.0, 0.0, 1.0, 1.0, 2, 0.9],
...     ]),
... ]

>>> result = sv.MeanAveragePrecision.from_tensors(predictions=predictions, targets=targets)
# MeanAveragePrecision(map=0.864625, map50=0.995, map75=0.80875, average_precisions=array([    0.73425,       0.995]))

```

Reply:
Those are simply average precisions per class. Not per threshold but already averaged.

@hardikdava is it what you intended?

Reply:
Ohh yeah, average_precision is already per class map values. Sorry for miscommunication. This is intended.